packages/create-app/src/lib/paths.ts

```diff
@@ -0,0 +1,150 @@
+ */
+
+import fs from 'fs-extra';
+import { dirname, resolve as resolvePath } from 'path';
+
+export type ResolveFunc = (...paths: string[]) => string;
+
+// Common paths and resolve functions used by the cli.
+// Currently assumes it is being executed within a monorepo.
+export type Paths = {
+  // Root dir of the cli itself, containing package.json
+  ownDir: string;
+
+  // Monorepo root dir of the cli itself. Only accessible when running inside Backstage repo.
+  ownRoot: string;
+
+  // The location of the app that the cli is being executed in
+  targetDir: string;
+
+  // The monorepo root package of the app that the cli is being executed in.
+  targetRoot: string;
+
+  // Resolve a path relative to own repo
+  resolveOwn: ResolveFunc;
+
+  // Resolve a path relative to own monorepo root. Only accessible when running inside Backstage repo.
+  resolveOwnRoot: ResolveFunc;
+
+  // Resolve a path relative to the app
+  resolveTarget: ResolveFunc;
+
+  // Resolve a path relative to the app repo root
+  resolveTargetRoot: ResolveFunc;
+};
+
+// Looks for a package.json that has name: "root" to identify the root of the monorepo
+export function findRootPath(topPath: string): string {
+  let path = topPath;
+
+  // Some sanity check to avoid infinite loop
+  for (let i = 0; i < 1000; i++) {
+    const packagePath = resolvePath(path, 'package.json');
+    const exists = fs.pathExistsSync(packagePath);
+    if (exists) {
+      try {
+        const data = fs.readJsonSync(packagePath);
+        if (data.name === 'root' || data.name.includes('backstage-e2e')) {
+          return path;
+        }
+      } catch (error) {
+        throw new Error(
+          `Failed to parse package.json file while searching for root, ${error}`,
+        );
+      }
+    }
+
+    const newPath = dirname(path);
+    if (newPath === path) {
+      throw new Error(
+        `No package.json with name "root" found as a parent of ${topPath}`,
+      );
+    }
+    path = newPath;
+  }
+
+  throw new Error(
+    `Iteration limit reached when searching for root package.json at ${topPath}`,
+  );
+}
+
+// Finds the root of the cli package itself
+export function findOwnDir() {
+  // Known relative locations of package in dist/dev
+  const pathDist = '..';
+  const pathDev = '../..';
+
+  // Check the closest dir first
+  const pkgInDist = resolvePath(__dirname, pathDist, 'package.json');
+  const isDist = fs.pathExistsSync(pkgInDist);
+
+  const path = isDist ? pathDist : pathDev;
+  return resolvePath(__dirname, path);
+}
+
+// Finds the root of the monorepo that the cli exists in. Only accessible when running inside Backstage repo.
+export function findOwnRootPath(ownDir: string) {
+  const isLocal = fs.pathExistsSync(resolvePath(ownDir, 'src'));
+  if (!isLocal) {
+    throw new Error(
+      'Tried to access monorepo package root dir outside of Backstage repository',
+    );
+  }
+
+  return resolvePath(ownDir, '../..');
+}
+
+export function findPaths(): Paths {
```

Comment:
Planning to move this entire util and file to a separate package, so let's duplicate this for now and deal with it later.